### PR TITLE
feat(module:icon): support add icon in feat modules & refactor

### PR DESCRIPTION
**Pull request by wendellhu95**
BREAKING CHANGE:

- NZ_ICON_DEFAULT_TWOTONE_COLOR is removed. Use NzGlobalConfigService instead.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by hsuanxyz** [comment]
@wendellhu95 测试挂了

File: components/icon/doc/index.en-US.md [hunk 1]
```diff
@@ -130,6 +130,19 @@ Let's assume that you deploy static assets under `https://mycdn.somecdn.com/icon
 
 Please call this in component's constructor or `AppInitService`.
 
+### Add Icons in Lazy-loaded Modules
+
+Sometimes, you want to import icons in lazy modules to avoid increasing the size of the main.js. You can use `NzIconModule.patch`.
+
+```ts
+@NgModule({
+  imports: [CommonModule, NzIconModule.patch([QuestionOutline])],
```
Comment (vthinkxie) [inline components/icon/doc/index.en-US.md]: maybe NzIconModule .forChild is better?

Comment (vthinkxie) [reply components/icon/doc/index.en-US.md]: and change { provide: NZ_ICONS, useValue: icons } to forRoot

Comment (wendellhu95) [reply components/icon/doc/index.en-US.md]: 增加了，已经在文档中去除了直接提供 NZ_ICONS 的使用方法。

**Comment by wendellhu95** [comment]
GitHub 这是怎么了，我已经 rebase 到了 master

File: components/icon/doc/index.en-US.md (hunk 1 above)
Comment (vthinkxie) [reply components/icon/doc/index.en-US.md]: patch的方法改成forChild?

File: components/icon/icons.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  LoadingOutline,
+  PaperClipOutline,
+  QuestionCircleOutline,
+  RightOutline,
+  StarFill,
+  SearchOutline,
+  StarFill,
+  UploadOutline,
+  UpOutline
+];
```
Comment (vthinkxie) [inline components/icon/icons.ts]: 这部分的图标是不是可以通过forChild直接在每个组件里面import

**Pull request merged**